skia/RNKeyEvent.cpp

```diff
@@ -0,0 +1,112 @@
+*/
+#include <folly/dynamic.h>
+#include <cxxreact/Instance.h>
+#include <cxxreact/JsArgumentHelpers.h>
+
+#include "RNKeyEvent.h"
+
+namespace facebook {
+namespace xplat {
+
+RNKeyEventModule::RNKeyEventModule() {}
+
+RNKeyEventModule::~RNKeyEventModule() {
+  stopObserving();
+}
+
+std::string RNKeyEventModule::getName() {
+  return "RNKeyEvent";
+}
+
+auto RNKeyEventModule::getMethods() -> std::vector<Method> {
+  return {
+      Method(
+          "addListener",
+          [&] (folly::dynamic args) {
+            RNS_LOG_DEBUG("addListener args "<< args[0]);
+            listenerCount_++;
+            if (listenerCount_ == 1) { 
+              startObserving();
+            }
+          }),// end of addListener lambda
+
+      Method(
+          "removeListeners",
+          [&] (folly::dynamic args) {
+            RNS_LOG_DEBUG(" removeListeners args "<< args[0]<<" listenerCount_ "<<listenerCount_);
+            int  removeCount = args[0].asInt();;
+            listenerCount_ = std::max(listenerCount_ - removeCount, 0);
+            if (listenerCount_ == 0) {
+              stopObserving();
+            }
+          }),
+  };
+}
+
+void RNKeyEventModule::startObserving(){
+    auto inputEventManager = react::RSkInputEventManager::getInputKeyEventManager();
+    if ( !inputEventManager ) {
+      RNS_LOG_ERROR("Unable to get RSkInputEventManager instance ");
+      return;
+    }
+    callbackId_ = inputEventManager->addKeyEventCallback(
+        [&](react::RSkKeyInput keyInput) {
+        // lambda for RSkinputEvent manager registation.-starting
+          if(keyInput.repeat_ == true){
+            repeatCount_++;
+          }
+          folly::dynamic eventPlayload = generateEventPayload(keyInput);
+          sendEventWithName(keyInput.action_ == RNS_KEY_Press?"onKeyDown":"onKeyUp",eventPlayload);
+          if(repeatCount_ > 0 && keyInput.action_ == RNS_KEY_Release){
+            folly::dynamic eventPlayload = generateEventPayload(keyInput,repeatCount_);
+            sendEventWithName("onKeyMultiple",eventPlayload);
+            repeatCount_ = 0;//rest back once you send the event.  
+          }
+        }
+    );//lambda for RSkinputEvent manager registation-end
+}
+folly::dynamic RNKeyEventModule::generateEventPayload(react::RSkKeyInput keyInput,unsigned int repeatCount){
+  
+  folly::dynamic eventPlayload = folly::dynamic::object();
+  if(repeatCount > 0 ){
+    eventPlayload["repeatcount"] = repeatCount;
+    //TODO character param need to update
+  }
+
+  eventPlayload["keyCode"] = (int) keyInput.key_;
+  eventPlayload["action"] = (int) keyInput.action_;
+  eventPlayload["pressedKey"] = (std::string) RNSKeyMap[keyInput.key_];
+  return eventPlayload;
+}
+void RNKeyEventModule::stopObserving(){
+  auto inputEventManager = react::RSkInputEventManager::getInputKeyEventManager();
+  if ( !inputEventManager ) {
+    RNS_LOG_ERROR("Unable to get RSkInputEventManager instance ");
+    return;
+  }
+  if(callbackId_ > 0){
```

Comment:
this check probably can be done at beginning itself to be call to get inputEventManager